src/borg/testsuite/helpers.py

```diff
@@ -607,6 +607,19 @@ def test_get_config_dir(monkeypatch):
         assert get_config_dir() == "/var/tmp"
 
 
+def test_get_config_dir_compat(monkeypatch):
+    """test that it works the same for legacy and for non-legacy implementation"""
+    monkeypatch.delenv("BORG_CONFIG_DIR", raising=False)
+    monkeypatch.delenv("XDG_CONFIG_HOME", raising=False)
+    assert get_config_dir(legacy=False) == get_config_dir(legacy=True)
+    # TODO fails on macOS: assert '/Users/tw/Library/Preferences/borg' == '/Users/tw/.config/borg'
+    monkeypatch.setenv("XDG_CONFIG_HOME", "/var/tmp/.config1")
+    assert get_config_dir(legacy=False) == get_config_dir(legacy=True)
+    # TODO fails on macOS: assert '/Users/tw/Library/Preferences/borg' == '/var/tmp/.config1/borg'
```

Comment:
looks like platformdirs does not respect it when XDG_CONFIG_HOME is set.

Reply:
on Linux, this works ok.

Reply:
on cygwin, this works ok.

https://github.com/platformdirs/platformdirs/issues/58#issuecomment-1416242459